app/src/lib/git/merge.ts

```diff
@@ -80,3 +80,10 @@ export async function mergeTree(
 
   return parseMergeResult(output)
 }
+
+/** Abort a mid-flight (conflicted) merge
+ *  @param repository where to abort the merge
+ */
+export async function abortMerge(repository: Repository): Promise<IGitResult> {
```

Comment:
Could you explain the decision to return an `IGitResult` over a `MergeResult`? Most of the git function I've come across get an `IGitResult` back from Dugite and are then parsed to return a type we use in the UI or store.

Reply:
mmmm, that's a good question. I'm honestly not sure what makes the most sense for this function to return. I don't think `IGitResult` will never have any useful information since we're not catching...perhaps it should just return `Promise<null>`.

does that make sense? the only other thing I think it could return is `true` but it would never return anything else.

Reply:
oh sorry i forgot to say that i didn't use `MergeResult` because this aborts a merge and none of the `MergeResult` types make sense for that case. I guess another option is to create a `MergeAbortResult` type and have it be either `Successful` or `Errored`.

I'm thinking `abortMerge` will be used rarely and only fail when the user has done something outside of github desktop so I was thinking it should let errors propagate up.

Reply:
`MergeResult` is a union type, so you would be able to extend it with a `MergeAborted` type.